layers/meta-balena-5x-owa/recipes-core/images/balena-image.inc

```diff
@@ -38,13 +38,14 @@ IMAGE_INSTALL:append = " \
                         linux-firmware-owasys \
                         wpa-supplicant \
                         udev-rules-owasys \
-                        system-maintenance-service \
                         rtc-tools \
                         quectel-cm \
                         switch-gsm \
                         st32-firmware \
                         chatscripts \
                         issue-owa \
+                        sync-time-rtc \
+                        owasys-hw-sn-file \
 "
```

Comment:
Okey, I'll see what can I do with the `owasys-hw-sn-file`. Now, about the `sync-time-rtc`.
I understand that BalenaOS is getting the time through internet, so no connectivity means that the system won't be able to set a correct date/time.
Regarding the owa5X, our board has an external RTC and a battery. The RTC can't be handled through kernel's drivers because it is implemented by the secondary onboard CPU. If we do not use our utilities we will just spoil our time keeping system capabilities, thus reducing the features that both our board and your BalenaOS can offer to potential clients. 
Is there any hook or place on the system that I can leverage to perform this `sync-time-rtc` operation in harmony with the rest of the system?

Thanks in advance

Reply:
@owzagj so, a high level overview of the balenaOS timekeeping framework is:

* Initially time is set from the image build time timestamp
* The system then checks whether the previous boot system time was stored in persistent storage and uses it to correct the time. The system time is stored in persistent storage when gracefully rebooting or shutting down.
* Logging starts after the last boot system time is set
* If an RTC is present, the system time will then be corrected using it.
* The system will then wait for a network connectivity event
* An HTTPs time synchronization server is then used to correct the time from HTTP headers timestamps, assuming the correct system time has not been set from the RTC previously
* Finally an NTP client is launched who takes care of keeping the time synchronized
* The NTP client is configured to synchronize every ~4h. There is a periodic healthcheck that retries a a time synchronization if the NTP servers are unreachable
* If the time gets unsynchronized, the NTP client service is restarted to correct failures

So, I understand your device has an RTC that can only be accessed via a secondary CPU, and that there is no kernel driver that can expose it via a standard `/dev/rtc` interface.

The less intrusive would be to replace the `recipes-core/systemd/timeinit/timeinit-rtc.sh` script with your own, so that it is called in the order established above.

Reply:
Hello, 

thanks for the explanation, much appreciated. I'm splitting the PR and creating a new one with only the replacement of `timeinit-rtc.sh` and the upgrade of `meta-owasys`

Reply:
In the meantime, could you please take a look at the new logo [PR](https://github.com/balena-io/contracts/pull/360) at contracts?

Thanks in advance!

Reply:
Hello,

I've created another [PR](https://github.com/balena-os/balena-owa5x/pull/46) removing system-mainentance, updated RTC date source and updating our layer meta-oways. Could you please review if everything is okay?


Thanks!

Reply:
It's done @owzagj, that PR will auto-merge once the CI tests pass.

Reply:
Thanks so much! I'm closing this PR as we are decided that won't go with the `owasys-sn-hw-file` recipe and nothing remains from this PR that's not tackled in others.